src/test/java/org/apache/brooklyn/cloudfoundry/location/CloudFoundryPaasLocationTest.java

```diff
@@ -39,25 +41,24 @@
     public void setUp() throws Exception {
         super.setUp();
         MockitoAnnotations.initMocks(this);
-        cloudFoundryPaasLocation = createCloudFoundryPaasLocation(client);
+        cloudFoundryPaasLocation = createCloudFoundryPaasLocation();
     }
 
     @Test
-    public void testSetUpClient() {
-        cloudFoundryPaasLocation.getClient();
+    public void testClientSetUp() {
         assertNotNull(cloudFoundryPaasLocation.getClient());
     }
 
-    @Test
-    public void testClientSingletonManagement() {
-        CloudFoundryClient client1 = cloudFoundryPaasLocation.getClient();
-        CloudFoundryClient client2 = cloudFoundryPaasLocation.getClient();
-        assertNotNull(client1);
-        assertEquals(client1, client2);
-    }
+    private CloudFoundryPaasLocation createCloudFoundryPaasLocation() {
+        Map<String, String> m = MutableMap.of();
+        m.put("user", "super_user");
+        m.put("password", "super_secret");
+        m.put("org", "secret_organization");
+        m.put("endpoint", "https://api.super.secret.io");
+        m.put("space", "development");
 
-    private CloudFoundryPaasLocation createCloudFoundryPaasLocation(CloudFoundryClient client) {
-        return new CloudFoundryPaasLocation(client);
+        return (CloudFoundryPaasLocation)
+                mgmt.getLocationRegistry().getLocationManaged("cloudfoundry", m);
```

Comment:
produces an error, because when the client is `SetUp` in init, the endpoint `https://api.super.secret.io` is not reachable:

`````
new CloudFoundryClient(
    new CloudCredentials(getConfig(CF_USER), getConfig(CF_PASSWORD)),
    getTargetURL(getConfig(CF_ENDPOINT)), //requires an reachable and existent CF platform
    getConfig(CF_ORG),
    getConfig(CF_SPACE), true);
``````

I do not think we should use an existent endpoint (such as `https://api.run.pivotal.io` ) because it is an unit test and the tests should be independent of the external environment. Then, I think we need to mock the `client`. I do not think we have to mock the CF platform (for example using mockito server) in order to simulate an reachable CF platform.

WDYT?